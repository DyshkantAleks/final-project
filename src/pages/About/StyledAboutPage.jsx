import styled from "styled-components";
import {device} from "../../styles/breakpoints/breakpoints";

export const ContactLink = styled.a`
font-size: 1.4rem;
line-height: 1.5;
text-decoration: underline;
color: inherit;
`;

export const TextContainer = styled.div`
flex-basis: 60%;
margin-right: 5rem;
`;
export const ContainerPage = styled.div`
display: flex;
flex-direction: column;
@media${device.tabletM}{
justify-content: space-between;
flex-direction: row;
}
`;
export const ContainerContactForm = styled.div`
border: .1rem solid #aecfd9;
box-shadow: 5px 5px 10px #aecfd9 ;
padding: 2rem;
@media${device.tabletM}{
width: 30rem;
margin: 0 auto;
align-self: flex-start;
}
@media${device.tabletM}{
margin: 0;
}
`;
export const ContactForm = styled.p`
font-size: 1.4rem;
line-height: 1.5;
font-weight: 500;
@media${device.tabletM}{

}
`;


export const Text = styled.p`
font-size: 1.4rem;
line-height: 1.5;
`

export const TextList = styled.ul`
font-size: 1.4rem;
line-height: 1.5;


`
export const TextListItem = styled.li`

`