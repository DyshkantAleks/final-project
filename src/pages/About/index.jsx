import React from 'react';
import {Header} from '../../commons/Header/Header';
import {AboutText} from "./AboutPageComponents/AboutText";
import { ContentContairer } from '../../components/Content/Content';

export const AboutPage = () => {
  return (
    <ContentContairer>
      <Header/>
      <h1>About page</h1>
      <AboutText/>
    </ContentContairer>
  )
}