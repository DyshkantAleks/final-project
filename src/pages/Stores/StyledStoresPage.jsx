import styled from "styled-components";

export const MapContainer = styled.div`
width: 100%;
height: 400px; 

`

export const Map = styled.iframe`

width: 100%;
height: 100%; 
// height: 100%;
frameborder: 0;  
allowfullscreen:; 
aria-hidden: false;
tabindex: 0;
`