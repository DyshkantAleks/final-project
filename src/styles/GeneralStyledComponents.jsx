import styled from 'styled-components';
import { device } from './breakpoints/breakpoints';

export const ContentContainer = styled.div`
    padding: 0 1.6rem;
    margin: 0 auto;
    
    @media ${device.tabletM}{
        padding: 0 2.4rem;
    }
    @media ${device.tabletL}{
        padding: 0 5rem;
    }
    
    @media ${device.desktop}{
        padding: 4rem 0;
        width: 120rem;
    }
`