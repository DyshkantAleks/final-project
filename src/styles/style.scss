:root {
  font-size: 10px !important; //надо important заменить при сборке
}

body {
  box-sizing: border-box;
  width: 100%;
  min-width: 320px;
  margin: 0;
  padding: 0;
  font-family: 'Montserrat', sans-serif !important;

  .ant-tooltip {
    z-index: 1200;
  }
}

a:hover {
  color: initial !important;
}