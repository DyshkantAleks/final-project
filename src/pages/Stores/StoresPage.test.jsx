import React from 'react';
import { shallow } from 'enzyme';

import { StoresPage } from './index';
import { ContentContainer } from '../../styles/GeneralStyledComponents';
import { ContainerPage, Text, TextContainer } from '../About/StyledAboutPage';
import { Map, MapContainer } from './StyledStoresPage';

const setUp = (props = {}) => shallow(<StoresPage {...props} />);

describe('StoresPage component', () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });
  it('should be render with general wrapper', () => {
    const generalWrapper = component.find(ContentContainer);
    expect(generalWrapper.length).toBe(1);
  });
  it('should be render with wrapper', () => {
    const wrapper = component.find(ContainerPage);
    expect(wrapper.length).toEqual(1);
  });
  it('should be render with text container', () => {
    const textContainer = component.find(TextContainer);
    expect(textContainer.length).toEqual(1);
  });
  it('should be render with text', () => {
    const text = component.find(TextContainer).find(Text);
    expect(text.length).toBe(3);
    expect(text.at(0).text()).toEqual('Для вашего удобства мы создали современный шоу-рум.');
    expect(text.at(1).text()).toEqual('В нашем шоу-руме представлены экспозиции современных брендов Италии, где вы можете посмотреть все товары, представленные у нас на сайте');
    expect(text.at(2).text()).toEqual('График работы: с понедельника по воскресенье с 10:00 до 20:00');
  });
  it('should render with map container', () => {
    const mapContainer = component.find(MapContainer);
    expect(mapContainer.length).toEqual(1);
  });
  it('should render with <Map />', () => {
    const map = component.find(Map);
    expect(map.prop('src')).toEqual('https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2541.651640662533!2d30.59143871518244!3d50.42896179679486!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x40d4c57deccaada1%3A0xa5e7cf7c9f8ff18b!2sDAN.IT!5e0!3m2!1sru!2sua!4v1603998516813!5m2!1sru!2sua');
  });
  it('snapshot <StoragePage />', () => {
    const generalWrapper = component.find(ContentContainer);
    expect(generalWrapper.debug()).toMatchSnapshot();
  });
});