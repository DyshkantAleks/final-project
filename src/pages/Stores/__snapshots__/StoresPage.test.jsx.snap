// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoresPage component snapshot <StoragePage /> 1`] = `
"<styled.div>
  <styled.div>
    <styled.div>
      <styled.p>
        Для вашего удобства мы создали современный шоу-рум.
      </styled.p>
      <styled.p>
        В нашем шоу-руме представлены экспозиции современных брендов Италии, где вы можете посмотреть все товары, представленные у нас на сайте
      </styled.p>
      <styled.p>
        График работы: с понедельника по воскресенье с 10:00 до 20:00
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.iframe src=\\"https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2541.651640662533!2d30.59143871518244!3d50.42896179679486!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x40d4c57deccaada1%3A0xa5e7cf7c9f8ff18b!2sDAN.IT!5e0!3m2!1sru!2sua!4v1603998516813!5m2!1sru!2sua\\" />
    </styled.div>
  </styled.div>
</styled.div>"
`;
